addons/survey/views/survey_templates_print.xml

```diff
@@ -16,11 +16,13 @@
                     <div role="form">
                         <fieldset disabled="disabled">
                             <t t-foreach='survey.question_and_page_ids' t-as='question'>
-                                <hr t-if="question.is_page and question != survey.page_ids[0]" />
-                                <div t-if="question.is_page" class="o_page_header">
-                                    <h1 t-field='question.title' />
-                                    <div t-if="question.description" t-field='question.description' class="oe_no_empty"/>
-                                </div>
+                                <t t-if="question.is_page">
```

Comment:
You also modified this part in the commit: "[FIX] survey: display only answered question in survey review mode"

Maybe it's related to that change?

This commit should probably not touch that part, and still unsure what it does.
I tested a random survey on the runbot and the layouts are the same.